qrenderdoc/Windows/ShaderViewer.cpp

```diff
@@ -1765,14 +1765,16 @@ void ShaderViewer::combineStructures(RDTreeWidgetItem *root, int skipPrefixLengt
       continue;
     }
 
-    // sort the children by offset, then name (in case all offsets are empty)
+    // Sort the children by offset, then global source var index, in case all offsets are empty.
+    // Using the global source var index allows resource arrays to be presented in index order
+    // rather than by name, so for example arr[2] comes before arr[10]
     std::sort(matches.begin(), matches.end(),
               [](const RDTreeWidgetItem *a, const RDTreeWidgetItem *b) {
                 VariableTag at = a->tag().value<VariableTag>();
                 VariableTag bt = b->tag().value<VariableTag>();
                 if(at.offset != bt.offset)
                   return at.offset < bt.offset;
-                return a->text(0) < b->text(0);
+                return at.globalSourceVar < bt.globalSourceVar;
```

Comment:
It's fine to add `globalSourceVar` as a preferred method of sorting, but please keep the text sorting as a final  fallback.

Reply:
Fixed